Swift 2.3

mattrubin opened the pull request:
Update the dependencies and configure the Xcode project so Guanaco can build with Xcode 8 and Swift 2.3.

This PR also uses Carthage to manage the dependency submodules, making it much easier to update dependencies to compatible versions.

mattrubin committed:
Use Carthage to manage submodules

mattrubin committed:
Update dependencies to support Swift 2.3

mattrubin committed:
Configure Guanaco.xcodeproj to build with Swift 2.3

mattrubin committed:
Configure Travis CI to build with Xcode 7.3, 8.0, and 8.1

mattrubin committed:
Disable code signing

Xcode 8.1: "Signing frameworks at build time is not recommended."

mattrubin committed:
Clean up FRAMEWORK_SEARCH_PATHS

This change removes some nonexistent search paths, silencing some "directory not found" linker warnings.

mattrubin committed:
Set low project deployment targets

This allows other projects which use Guanaco more flexibility in their own deployment targets.

mattrubin committed:
Update to Xcode 8.1 recommended settings

mattrubin commented:
The Travis builds seem to be flaky. They are failing here, but passing [for all of these commits on `mattrubin/swift-2.3`](https://github.com/mattrubin/Guanaco/commits/c7c186b3c93f655d01ddf44b08232845ccd2fb66).

modocache commented:
Hey @mattrubin! I added you as a collaborator so that you don't have to wait for me to review your work. That being said, this LGTM! Thanks!!

mattrubin committed:
Configure Travis CI to build with Xcode 8.1 only

Xcode 7.3 builds on Travis were repeatedly failing because xcodebuild would build dependencies for the wrong platform. (rdar://20490378, rdar://22008701, rdar://22816454)
Xcode 8.0 builds on Travis were occasionally failing because xcodebuild would exit with status 65 and no explanation of the failure.

mattrubin commented:
Thanks, @modocache!
